compiler/rustc_middle/src/ty/error.rs

```diff
@@ -263,7 +263,20 @@ impl<'tcx> Ty<'tcx> {
             }
             ty::Slice(ty) if ty.is_simple_ty() => format!("slice `{}`", self).into(),
             ty::Slice(_) => "slice".into(),
-            ty::RawPtr(_) => "*-ptr".into(),
+            ty::RawPtr(tymut) => {
+                let tymut_string = match tymut.mutbl {
+                    hir::Mutability::Mut => tymut.to_string(),
+                    hir::Mutability::Not => format!("const {}", tymut.ty),
+                };
+                if tymut_string != "_"
+                    && (tymut.ty.is_simple_text() || tymut_string.len() < "raw pointer".len())
```

Comment:
oh wait, I see. You're using the length of the rendered string here.

Reply:
> `tymut_string.len() < "raw pointer".len())`

Do we need to do this? I think `is_simple_text` is sufficient

Reply:
It's probably useless, yeah. Since we already have `const` or `mut` prefixed, the amount of chars that fits within "raw pointer" is very small, and should already be covered by `is_simple_text`. But maybe it would make sense to raise the max length

Reply:
I can test it out on the ui tests

Reply:
Thanks

Reply:
It made no difference to the UI tests

Reply:
Which I would guess is mostly caused by `raw pointer` being such a small limit

Reply:
I just removed it, this should be good enough.